lib/active_storage/service/mirror_service.rb

```diff
@@ -26,9 +26,8 @@ def primary_service
     end
 
     def perform_across_services(method, *args)
-      # FIXME: Convert to be threaded
       services.collect do |service|
-        service.public_send method, *args
-      end
+        Thread.new { service.public_send method, *args }
+      end.map(&:join)
```

Comment:
I've changed the code to use thread pool executor, and now it looks like this:
```ruby
def perform_across_services(method, *args)
  pool = Concurrent::ThreadPoolExecutor.new(max_threads: @services.count)
  # pool is local only for the example purpose
  services.each do |service|
    pool.post { service.public_send method, *args }
  end
end
```
But this causes troubles with the test because threads do not join.
I can make them pass by sleeping for a while, but this does not seem like a good solution to the problem. Can you advise me how to handle this?